packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,244 @@
+ */
+export interface MotionCanvasCorsProxyPluginOptions {
+  /**
+   * Set which types of resources are allowed by default.
+   *
+   * @remarks
+   * Catchall on the right side is supported.
+   * Pass an empty Array to allow all types of resources, although this is not recommended.
+   *
+   * @default ["image/*", "video/*"]
+   */
+  allowedMimeTypes?: string[];
+  /**
+   * Set which hosts are allowed
+   *
+   *
+   * @remarks Note that the host is everything to the left of the first `/`, and to the right of the protocol `https://`
+   * Allowlist is not used by default, although you should consider setting up just the relevant hosts.
+   *
+   * Note that `api.iconify.design` is **always** enabled, as it is used by the Icon-Component.
+   * It gets added by the Middleware internally.
+   */
+  allowListHosts?: string[];
+}
+
+// Add Hosts required for core functionality to work
+const predefinedAllowlist = [
+  // Icons
+  'api.iconify.design',
+];
+
+export function motionCanvasCorsProxy(
+  middleware: Connect.Server,
+  config: MotionCanvasCorsProxyPluginOptions | undefined = {},
+) {
+  // Set the default config if no config was provided
+  config.allowedMimeTypes ??= ['image/*', 'video/*'];
+  config.allowListHosts ??= [];
+
+  config.allowListHosts.push(...predefinedAllowlist);
+
+  // Check the Mime Types to have a correct structure (left/right)
+  // not having them in the correct format would crash the Middleware
+  // further down below
+  if ((config.allowedMimeTypes ?? []).some(e => e.split('/').length !== 2)) {
+    throw new Error(
+      "Invalid config for Proxy:\nAll Entries must have the following format:\n 'left/right' where left may be '*'",
+    );
+  }
+
+  middleware.use((req, res, next) => {
+    if (!req.url || !req.url.startsWith('/cors-proxy/')) {
+      // url does not start with /cors-proxy/, so this
+      // middleware does not care about it
+      return next();
+    }
+
+    // For now, only allow GET Requests
+    if (req.method !== 'GET') {
+      return writeError('Only GET Requests are allowed', res, 405);
+    }
+
+    let sourceUrl: URL;
+    try {
+      sourceUrl = extractDestination(req.url);
+    } catch (err) {
+      return writeError(err as any, res);
+    }
+
+    if (
+      !isReceivedUrlInAllowedHosts(sourceUrl.hostname, config.allowListHosts)
+    ) {
+      return writeError(
+        `Blocked by Proxy: ${sourceUrl.hostname} is not on Hosts Allowlist`,
+        res,
+      );
+    }
+
+    // Get the resource, do some checks. Throws an Error
+    // if the checks fail. The catch then writes an Error
+    return tryGetResource(res, sourceUrl, config).catch(error => {
+      writeError(error, res);
+    });
+  });
+}
+
+/**
+ * Unwrap the destination from the URL.
+ *
+ * @param url - the entire URL with the `/cors-proxy/` prefix and containing the url-Encoded Path
+ * @returns the URL that needs to be called
+ *
+ * @remarks Throws an Error if the value could not be unwrapped.
```

Comment:
Remarks should always come second.
Also missing commas/capitalization.
```suggestion
 * Unwrap the destination from the URL.
 *
 * @remarks 
 * Throws an Error if the value could not be unwrapped.
 *
 * @param url - the entire URL with the `/cors-proxy/` prefix and containing the
 * url-Encoded Path.
 *
 * @returns The URL that needs to be called.
```